### PR TITLE
relative paths for icon and version in spec file #3333

### PR DESCRIPTION
**Pull request by tallforasmurf**
The code changes are I believe complete for this issue. There are problems with the manual, but I don't want to put doc changes into this pull request. After it is accepted I will open a new issue specifically to better document these various command parameters and their related spec file values.

### CODE REVIEW

**Comment by htgoebel** [comment]
Thanks for the pull-request.

All test-runs are failing. I'm afraid, there is a small but serious error in the code, see https://travis-ci.org/pyinstaller/pyinstaller/jobs/362774665#L3061

**Comment by tallforasmurf** [comment]
I am mystified. I didn't change anything in the PYZ section, only EXE. And it was working fine in my tests.

Well, wait, one change I did make that could possibly be related is, each class had its own copy of `from ..config import CONF` and I moved that up to the top with the other global imports. Perhaps that gets it too early, before "workpath" has been defined? (But it definitely has "specpath" defined when EXE runs...) OK, I will put the imports back the way they were and hope.

commit: https://github.com/tallforasmurf/pyinstaller/commit/a86349e2fdb523bc6d416fdb3bd0df1a7bfeadb3

**Comment by tallforasmurf** [comment]
Wait, don't test yet. I just remembered why I made that change, it wasn't merely for tidyness. Must check.

File: PyInstaller/building/api.py [hunk 1]
```diff
@@ -406,6 +406,11 @@ def __init__(self, *args, **kwargs):
                 self.toc.append(("pyi-windows-manifest-filename " + manifest_filename,
                                  "", "OPTION"))
 
+            if self.versrsrc :
+                if not os.path.isabs(self.versrsrc) :
```
[the remark this answers is not in the document]
Comment (htgoebel) [reply PyInstaller/building/api.py]: Also there are space in front of the colons.

**Comment by htgoebel** [comment]
@tallforasmurf May I ask you to clean up the pull-request and fix the open coding issues. I would like to merge this soon. Thanks.

Please reword your **commit messages** to comply to our [Commit Message Rules](https://pyinstaller.readthedocs.io/en/latest/development/commit-messages.html#commit-messages) - basically remove the issue number, which was required by an now outdated rule.

You also need to submit a [**changelog entry**](https://pyinstaller.readthedocs.io/en/latest/development/changelog-entries.html) so our users can learn about your change. (This is a new requirement since last Septembe to speed up releasing.)

**Comment by htgoebel** [comment]
@tallforasmurf pinging

**Comment by htgoebel** [comment]
@tallforasmurf :-(((((

**Comment by htgoebel** [comment]
@tallforasmurf So I did this work for you to get this finished at last. Please check the changelog-entries wither the description is correct.

**Pull request merged**